src/main/java/gregtech/client/renderer/RenderSetup.java

```diff
@@ -0,0 +1,30 @@
+package gregtech.client.renderer;
+
+import net.minecraft.client.renderer.BufferBuilder;
+import net.minecraftforge.fml.relauncher.Side;
+import net.minecraftforge.fml.relauncher.SideOnly;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Object representation of GL setup code. Any recurring render setup / cleanup code should probably go here.<p/>
+ * During render, render calls with identical render setup instance will be drawn in a batch.
+ * Providing proper equality checks and hashes is recommended for non-singleton render setup implementations.
+ */
+@SideOnly(Side.CLIENT)
+public interface RenderSetup {
```

Comment:
should this not be called `IRenderSetup`

Reply:
uh yeah probably